pcap.c

```diff
@@ -3152,6 +3152,21 @@ pcap_datalink_val_to_description(int dlt)
 	return (NULL);
 }
 
+const char *
+pcap_datalink_val_to_description_or_dlt(int dlt)
+{
+        static char unkbuf[40];
+        const char *description;
+
+        description = pcap_datalink_val_to_description(dlt);
+        if (description != NULL) {
+                return description;
+        } else {
+                snprintf(unkbuf, sizeof(unkbuf), "DLT %u", dlt);
```

Comment:
Use `pcap_snprintf()` instead; that works on more platforms.

Reply:
Thanks!

Reply:
> tcpdump: no VLAN support for Linux cooked.

Not `Linux cooked v1` ?

Reply:
> > tcpdump: no VLAN support for Linux cooked.
> 
> Not `Linux cooked v1` ?

If you apply this change to a version prior to 8cff296dc7c321c76933359d586dbde5b580ce8c, it'll be "Linux cooked".  If you apply it to 8cff296dc7c321c76933359d586dbde5b580ce8c or later, it'll be "Linux cooked v1".

Reply:
Sorry, my test environment is "libpcap 1.9.0 plus patches", not master.  Thanks for clarifying, Guy.  (I am looking forward to Linux cooked v2!)